### PR TITLE
Monkey rumble

### PR DESCRIPTION
**Pull request by gc**
### Other checks:

-   [ ] I have tested all my changes thoroughly.

### CODE REVIEW

File: src/commands/Minion/mmmr.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+**Fights Done:** ${scores.MadMarimbosMonkeyRumble}
+**Unique Monkeys Fought:** ${
+				msg.author.settings.get(UserSettings.MonkeysFought).length
+			}/${TOTAL_MONKEYS.toLocaleString()}
+**Greegree Level:** ${tier.greegree.name} - ${tier.id}/${monkeyTiers.length.toLocaleString()}`
+		);
+	}
+
+	async buy(msg: KlasaMessage, [input = '']: [string]) {
```
Comment (imgidedin) [inline src/commands/Minion/mmmr.ts:88]: Missing `@requiresMinion` decorator

File: src/commands/Minion/mmmr.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+		const buyable = buyables.find(i => stringMatches(input, i.item.name));
```
Comment (imgidedin) [inline src/commands/Minion/mmmr.ts:89]: Buyables is not using the `buyables.aliases` defined.

File: src/commands/Minion/mmmr.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+			oneAtTime: true,
+			altProtection: true,
+			categoryFlags: ['minion', 'pvm', 'minigame'],
+			description: 'Sends your minion to do barbarian assault, or buy rewards and gamble.',
+			examples: ['+barbassault [start]'],
+			subcommands: true,
+			usage: '[start|level|buy] [buyableOrGamble:...string]',
+			usageDelim: ' ',
+			aliases: ['mr', 'mmr', 'rumble']
+		});
```
Comment (imgidedin) [inline src/commands/Minion/mmmr.ts:70]: Still has the barbarian assault information

File: src/tasks/minions/minigames/monkeyRumbleActivity.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+				loot.add('Marimbo statue');
+			}
+			files.push(
+				await monkeyHeadImage({
+					monkey,
+					content: unique.has('Monkey egg')
+						? 'You are strong warrior. Take this monkey egg, raise him well and take care of him or I find you and crush you.'
+						: randArrItem(gotUniqueMessages)
+				})
+			);
```
Comment (imgidedin) [inline src/tasks/minions/minigames/monkeyRumbleActivity.ts:69]: This seems unnecessary, as for the return message, the first element is sent. Maybe do `if(files.length === 0) files.push(...);`. That will save time generating _n_ amount of chatheads that will never be used.

**Review by themrrobert — CHANGES_REQUESTED** [review]

File: src/commands/Minion/mmmr.ts [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+		const boosts = [];
+
+		const fightDuration = Time.Minute * 9;
+		const quantity = Math.floor(msg.author.maxTripLength(Activity.MonkeyRumble) / fightDuration);
+		let duration = quantity * fightDuration;
+		let chanceOfSpecial = Math.floor(300 * (6 - monkeyTierOfUser(msg.author) / 2));
+		const monkeysToFight: Monkey[] = [];
+		for (let i = 0; i < quantity; i++) {
+			monkeysToFight.push(getRandomMonkey(monkeysToFight, chanceOfSpecial));
```
[the remark this answers is not in the document]
Comment (themrrobert) [reply src/commands/Minion/mmmr.ts:192]: Actually this looks like it's not actually excluding all previously fought monkeys like I initially thought. So you can disregard

**Pull request merged**